massive restructure

sachiniyer opened the pull request:
This is opinionated, but when working on issue #18 it became apparent (at least to me) that the wiki as a whole needed some restructuring.

I have divided it into 
- Academics
- Opportunities
- Social

There is not much content change (just a few additions), just a restructure.

I also added some other stuff
- Mermaid Diagrams support (for the diagram on the Opportunities page) with theming compliance
- Home page tiles to the different sections
- Software Acknowledgement

Again opinionated, and perhaps uncalled for. However, because the activity in this repo is limited, I think a restructure right now should be okay.

sachiniyer commented:
Let me figure out why the github workflow is failing before review.

sachiniyer commented:
Okay I have merged with main, passed the linter, and passed the test build.

I am ready for review @aminoa

aminoa commented:
Hey, I finally got a chance to take a look. I really appreciate the improved diagrams, home page tiles, and software acknowledgement. I did break out the opportunities section into two categories: one for career, the other still called opportunities (basically representing the internal/external opportunities at NYU). So it's now like this.

Academics (same as before)

Career
- Internships
- Fellowships

Opportunties
- Professor Run
	- Labs
	- Research
- Student Run
	- Clubs
	- VIP
- Projects 
 
Social (same as before)

If you disagree with this, I can revert this and we can come up with a different structure. My goal was to limit the folds to only one layer to make the site easier to navigate. Also, when the structure is decided, I can make new images instead of using the DALLE ones (the social one is a little bit creepy lol).

sachiniyer commented:
## Splitting Opportunities and Career


I was trying to represent the opportunities in two different ways:
1. Based off of internal to NYU vs external to NYU:

    Represented in the TOC by Internal and External
3. Based off of goal - jobs vs higher education

    Represented in the diagram on the Opportunities landing page

I'm not opposed to a restructure. If the goal was to keep the TOC one level deep, I would flatten the Professor Run and Student Run section.

The new structure could look like (that maintains one level deep)

 - Opportunities (with diagram of goals out of college)
    - Internal to NYU
      - Labs - with marking for professor run
      - Research - with marking for professor run
      - Clubs - with marking for student
      - VIP - with marking for student
    - External to NYU
      - Internships
      - Fellowships


## New Images

Lol sure, no opposition.

If we do change the top structure, we may also want to add another widget specifically to the [Feature List](https://github.com/BUGS-NYU/nyu-cs-wiki/blob/0c33e356e7189b0abd25f2ad020e06e510655c6b/src/components/HomepageFeatures/index.jsx#L5C1-L6C1).

sachiniyer commented on src/components/HomepageFeatures/index.js:
```diff
@@ -4,45 +4,47 @@ import styles from './styles.module.css';
 
 const FeatureList = [
```
If we want to change the top structure, we will need to add a widget here

aminoa commented:
reverted the new header change; I broke out fellowships and internships into separate folds in the opportunities section since I suspect there will be quite a few more pages for both categories. Followed your suggestion for putting all the internal opportunities grouped together (I put tags for student and professor though if you want to have it in the title of each page, I wouldn't mind). 

Also here are the new images:

![image](https://github.com/BUGS-NYU/nyu-cs-wiki/assets/28660350/07863058-833f-4378-af91-e6572f9d882c)

aminoa commented:
also seems to fix #2 as well

sachiniyer commented:
This looks good to me. I just made a few small changes

1. Changed "Internal" to "NYU Internal" to make the meaning clearer
2. Changed the sidebar position to keep "NYU Internal" at the top (this made the most visual sense to me
3. Changed the ordering inside of "NYU Internal" to have Research, Labs, Clubs, VIP which also made the most sense to me
4. Added a small description to the top page

I would propose merging if this looks good to you. I think it is a good settling place.

Also I love the new images. Way better than what I did.

sachiniyer commented:
Also just quickly renamed the icon folder as the same images were in high and low res (so there is no point in having distinct folders anymore)

aminoa commented:
Yep, looks great. I appreciate your work on this PR!
